File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.rs
use std::collections::HashMap;
impl Solution {
    pub fn find_winners(matches: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let mut result: HashMap<i32, i32> = HashMap::new();

        for m in &matches {
            let winner = m[0];
            let loser = m[1];
            let w_cnt = result.entry(winner).or_insert(0);
            let l_cnt = result.entry(loser).or_insert(0);
            *l_cnt += 1;
        }

        let mut answer = vec![vec![], vec![]];
        
        for (player, count) in result {
            match count {
                0 => answer[0].push(player),
                1 => answer[1].push(player),
                _ => continue,
            }
        }
        
        for i in 0..=1 {
            answer[i].sort();
        }

        return answer;
        
    }
}

File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.py
from collections import defaultdict
class Solution:
    def findWinners(self, matches: List[List[int]]) -> List[List[int]]:
        result = defaultdict(int)
        participants = set()
        for winner, loser in matches:
            result[loser] += 1
            participants.add(loser)
            participants.add(winner)
        
        answer = [[], []]
        for player in participants:
            if result[player] == 0:
                answer[0].append(player)
            elif result[player] == 1:
                answer[1].append(player)
        for el in answer:
            el.sort()
        return answer
            